tools/cmake/mbed_toolchain.cmake

```diff
@@ -7,17 +7,9 @@ function(mbed_generate_options_for_linker target output_response_file_path)
         "$<TARGET_PROPERTY:${target},INTERFACE_COMPILE_DEFINITIONS>"
     )
 
-    # Remove macro definitions that contain spaces as the lack of escape sequences and quotation marks
-    # in the macro when retrieved using generator expressions causes linker errors.
-    # This includes string macros, array macros, and macros with operations.
-    # TODO CMake: Add escape sequences and quotation marks where necessary instead of removing these macros.
+    # Append -D to all macros and quote them as we pass these as response file to cxx compiler
     set(_compile_definitions
-       "$<FILTER:${_compile_definitions},EXCLUDE, +>"
-    )
-
-    # Append -D to all macros as we pass these as response file to cxx compiler
-    set(_compile_definitions
-        "$<$<BOOL:${_compile_definitions}>:-D$<JOIN:${_compile_definitions}, -D>>"
+        "$<$<BOOL:${_compile_definitions}>:'-D$<JOIN:${_compile_definitions},' '-D>'>"
```

Comment:
@0xc0170 I've tested it with sockets and mesh, both worked fine with both GCC_ARM and ARM compilers. And this also solves problems with macros with spaces inside (e.g. `'-DMBED_CONF_LORA_APPLICATION_EUI={0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}'`), so I removed the previous workaround.

Hopefully it's a permanent fix, let's run CI to test it on more examples!